crypto/src/test_utils/test_conversation/mod.rs

```diff
@@ -0,0 +1,294 @@
+        let mut member_guard = member
+            .transaction
+            .conversation(&self.id)
+            .await
+            .map_err(RecursiveError::transaction("getting conversation by id"))?;
+        let mut other_guard = other_member
+            .transaction
+            .conversation(&self.id)
+            .await
+            .map_err(RecursiveError::transaction("getting conversation by id"))?;
+        let msg = b"Hello other";
+        let encrypted = member_guard
+            .encrypt_message(msg)
+            .await
+            .map_err(RecursiveError::mls_conversation("encrypting message; member -> other"))?;
+        let decrypted = other_guard
+            .decrypt_message(encrypted)
+            .await
+            .map_err(RecursiveError::mls_conversation("decrypting message; other <- member"))?
+            .app_msg
+            .ok_or(TestError::ImplementationError)?;
+        assert_eq!(&msg[..], &decrypted[..]);
+        // other --> member
+        let msg = b"Hello member";
+        let encrypted = other_guard
+            .encrypt_message(msg)
+            .await
+            .map_err(RecursiveError::mls_conversation("encrypting message; other -> member"))?;
+        let decrypted = member_guard
+            .decrypt_message(encrypted)
+            .await
+            .map_err(RecursiveError::mls_conversation("decrypting message; member <- other"))?
+            .app_msg
+            .ok_or(TestError::ImplementationError)?;
+        assert_eq!(&msg[..], &decrypted[..]);
+        Ok(())
+    }
+
+    /// Check if one member can exchange application messages with another
+    pub(crate) async fn can_talk(&self, member: &SessionContext, other_member: &SessionContext) -> bool {
+        self.try_talk(member, other_member).await.is_ok()
+    }
+
+    /// Get the current acting member, i.e., the member on whose behalf the following operation will be performed.
+    /// After each operation, this will reset to the creator.
+    pub fn actor(&self) -> &SessionContext {
+        self.members[self.actor_index()]
+    }
+
+    fn actor_index(&self) -> usize {
+        self.actor_index.unwrap_or_default()
+    }
+
+    /// Execute the next operation on bahalf of the provided member.
+    /// This will reset to the conversation creator once the operation is distributed via
+    /// [OperationGuard::notify_members] or finished via [OperationGuard::finish].
+    pub async fn acting_as(mut self, actor: &SessionContext) -> Self {
+        let actor_index = self.member_index(actor).await;
+        self.actor_index = Some(actor_index);
+        self
+    }
+
+    /// All members of this conversation.
+    ///
+    /// The creator is always the first member returned (if they're still a member).
+    pub fn members(&self) -> impl Iterator<Item = &SessionContext> {
+        self.members.iter().copied()
+    }
+
+    /// Convenience function to get the mls transport of the creator.
```

Comment:
I guess in many cases the actor will be creator, but not always, so I think this should be "... transport of the actor".